Tools/i18n/pygettext.py

```diff
@@ -730,15 +595,9 @@ class Options:
             fp = open(filename, 'rb')
             closep = 1
         try:
-            eater.set_filename(filename)
-            try:
-                tokens = tokenize.tokenize(fp.readline)
-                for _token in tokens:
-                    eater(*_token)
-            except tokenize.TokenError as e:
-                print('%s: %s, line %d, column %d' % (
-                    e.args[0], filename, e.args[1][0], e.args[1][1]),
-                    file=sys.stderr)
+            module_tree = ast.parse(fp.read())
```

Comment:
Since we need all file content at once instead of reading it line by line, we can simplify reading. Read it by
```
source = sys.stdin.buffer.read()
```
and
```
with open(filename, 'rb') as fp:
    source = fp.read()
```
and get rid of `closep` and `try`/`finally`.

Reply:
Done :)